src/Components/Web.JS/src/Platform/WebAssemblyResourceLoader.ts

```diff
@@ -0,0 +1,195 @@
+import { toAbsoluteUri } from '../Services/NavigationManager';
+
+export class WebAssemblyResourceLoader {
+  private usedCacheKeys: { [key: string]: boolean } = {};
+  private networkLoads: { [name: string]: LoadLogEntry } = {};
+  private cacheLoads: { [name: string]: LoadLogEntry } = {};
+
+  static async initAsync(): Promise<WebAssemblyResourceLoader> {
+    const bootConfigResponse = await fetch('_framework/blazor.boot.json', {
+      method: 'GET',
+      credentials: 'include'
+    });
```

Comment:
Can we make sure we always download blazor.boot.json with cache: 'reload' or another mechanism? Right now, if a server incorrectly caches blazor.boot.json we are busted I believe.

Reply:
Yes, good point!

My gut preference is for `cache: 'no-cache'` so that a well-configured server is allowed to return a 304. If we set `cache: 'reload'` then we're punishing people who did configure their server properly. People who configure their servers in an invalid way (e.g., returning headers to cache for a specific time period regardless of content changes) will have trouble but that's also true for all their other static resources, `index.html`, etc.

I'll do that now, but please let me know if you think it really should be `reload`.

Reply:
I think 'no-cache' is fine. People can issue a request for it themselves if their server is not caching things right to force the update.

Reply:
Great, `no-cache` it is.